### ISSUE TITLE
Add thresholds to pain

### ISSUE THREAD

**Issue by fairyarmadillo** [CleverRaven/Cataclysm-DDA#72852]
### Is your feature request related to a problem? Please describe.

Currently, pain can balloon out of control as you encounter random minor damage in the world, which especially with the new and dramatically worse pain penalties can easily lead to a death spiral. This doesn't make a ton of sense - someone slapping your arm 100 times is not going to put you in an equivalent amount of pain to being shot in the stomach.

### Solution you would like.

Damage should only add to pain if it's under a certain amount, according to the damage dealt. This could also be extended to non-damage sources of pain, like effects and ie dragging furniture, though it would need to be implemented on a case-by-case basis for hardcoded stuff like that.

For example (this is just off the top of my head and these numbers should not actually be used):

To break 20 pain, you need to take over 10 damage from a single source, or have the target limb at <75% hp.
To break 50 pain, you need to take over 20 damage from a single source, or have the target limb at <50% hp.
To break 75 pain, you need to take over 30 damage from a single source, or have the target limb at <25% hp.
To break 100 pain, you need to take over 40 damage from a single source, or have the target limb at <10% hp.
To break 120 pain, you need to take over 50 damage from a single source, or the target limb must be broken.

### Describe alternatives you have considered.

This can probably vary by limb and by damage type, with acid and fire breaking pain thresholds more easily than cuts and bruises. The head should feel more pain than other parts, and some mutants would vary as well - a spider or a squid can lose an arm without completely succumbing to pain, for instance. But that can all be added later once the basic system is in place.

### Additional context

#72687 seems way overtuned, especially for early game characters who are taking a lot of chip damage, and this would help them a lot.

**Comment by I-am-Erk** [comment]
I think you might be on to something here, but it would be very important to balance it such that it doesn't remove the threat of smaller injuries entirely. Would need to run a bunch of numbers and see what consequences could arise.

**Comment by Nebnis** [comment]
For small injuries could probably add a timer to break the threshold. Let's say, getting a small cut won't do it but getting 10 small cuts in less than xx seconds would break the threshold. 

Fairy proposal seems really good for all this too.

**Comment by I-am-Erk** [comment]
That is basically what the wound system is for

**Comment by fairyarmadillo** [comment]
That's also handled in the proposal. See the bit about overall limb damage breaking thresholds.

**Comment by Rocket-F-1024** [comment]
I think it might be better to directly use the pain that could have been caused instead of the damage to calculate the pain that was finally caused, since damage and pain are not exactly equivalent (e.g. poisoning, acid)
In addition, medications might also be able to have an impact when it comes to pain (e.g., decrease/increase pain gain, increase/decrease pain tolerance)
